src/tests/functional/shared_test_classes/include/shared_test_classes/base/ov_subgraph.hpp

```diff
@@ -65,6 +64,11 @@ class SubgraphBaseTest : public ov::test::TestsCommon {
     ov::CompiledModel compiledModel;
     ov::InferRequest inferRequest;
 
+    // to provide correct inputs for reference function
+    std::map<std::shared_ptr<ov::op::v0::Parameter>, std::shared_ptr<ov::op::v0::Parameter>> matched_parameters;
+    precisions_map convert_precisions = {{ ov::element::bf16, ov::element::f32 },
+                                         { ov::element::f16, ov::element::f32 }};
```

Comment:
Why do you introduce the same logic as we had in legacy API? Shouldn't we compare original precisions?

Reply:
Could you please share a link? In other case we will have an accuracy problem for example for `fp16` case on CPU